src/components/share-button/share-button.component.ts

```diff
@@ -105,24 +73,25 @@ export class ShareButtonComponent implements AfterViewInit {
                 this.popUpClosed.emit(this.button.provider);
             }
         }, 200);
-    }
+    };
 
+    shareButton() {
+        this.btn = this.renderer.createElement(this.self, 'button');
+        this.renderer.setElementProperty(this.btn, 'innerHTML', this.button.template);
+        this.renderer.setElementAttribute(this.btn, 'class', this.button.classes);
+        this.renderer.setElementProperty(this.btn, 'onclick', this.share);
+    }
 
-    nFormatter(num, digits) {
-        let si = [
-            { value: 1E18, symbol: 'E' },
-            { value: 1E15, symbol: 'P' },
-            { value: 1E12, symbol: 'T' },
-            { value: 1E9, symbol: 'G' },
-            { value: 1E6, symbol: 'M' },
-            { value: 1E3, symbol: 'K' }
-        ], rx = /\.0+$|(\.[0-9]*[1-9])0+$/, i;
-        for (i = 0; i < si.length; i++) {
-            if (num >= si[i].value) {
-                return (num / si[i].value).toFixed(digits).replace(rx, '$1') + si[i].symbol;
-            }
-        }
-        return num.toFixed(digits).replace(rx, '$1');
+    shareCount() {
+        this.sbService.count(this.button.provider, this.url)
+            .subscribe(shareCount => {
+                if (shareCount) {
+                    let counter = this.renderer.createElement(this.self, 'span');
```

Comment:
Hi, 

I wasn't aware of theses issues, in particular with the 3rd option.
If that was the only reason for manipulating the DOM,  i think we can go full template and still benefit from change detection, by using [ChangeDetectorRef.markForCheck()](https://angular.io/docs/ts/latest/api/core/index/ChangeDetectorRef-class.html).  :)

See [this interesting article](https://blog.thoughtram.io/angular/2016/02/22/angular-2-change-detection-explained.html) by **Pascal Precht**.

Regarding tests, the problem with `ElementRef`, is that it cannot be directly injected in test environment, i have to do ugly things like creating a Mock class that extends ElementRef, and used it like this:

```
class MockElementRef extends ElementRef{}

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [HttpModule, JsonpModule],
      declarations: [ShareButtonComponent],
      providers: [WindowService, ShareButtonsService, Renderer, { provide: ElementRef, useClass: MockElementRef }]
    })
      .compileComponents();
  }));
```